src/Shared/Components/ComponentDescriptorSerializer.Deserialize.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text.Json;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // ComponentDescriptorSerializer is an abstraction shared by both MVC and Blazor
+    // This file holds the definition used by Blazor to deserialize descriptors.
+    internal partial class ComponentDescriptorSerializer
+    {
+        public ComponentDescriptorSerializer(
+            IDataProtectionProvider dataProtectionProvider,
+            ILogger<ComponentDescriptorSerializer> logger,
+            RootComponentTypeCache rootComponentTypeCache)
+        {
+            DataProtector = dataProtectionProvider.CreateProtector(DataProtectionProviderPurpose);
+            Logger = logger;
+            RootComponentTypeCache = rootComponentTypeCache;
+        }
+
+        public ILogger<ComponentDescriptorSerializer> Logger { get; }
+
+        public RootComponentTypeCache RootComponentTypeCache { get; }
+
+        public bool TryDeserializeComponentDescriptorCollection(string serializedComponentRecords, out List<ComponentDescriptor> descriptors)
+        {
+            var markers = JsonSerializer.Deserialize<IEnumerable<ComponentMarker>>(serializedComponentRecords, _jsonSerializationOptions);
+            descriptors = new List<ComponentDescriptor>();
+            int lastSequence = -1;
+
+            var previousInstance = new ComponentDescriptorInstance();
```

Comment:
This is why I suggested a btter name for this class. Here we have `ComponentDescriptor` and `ComponentDiscriptorInstance` getting to know each other.

What about ParsedComponentMarker since ComponentMarker is taken?

Reply:
I renamed this and all its friends to ServerComponent and ServerComponentSerializer and so on.